### PR TITLE
Upgrade obeam (support BigInt)

### PR DESCRIPTION
**Pull request by amutake**

### CODE REVIEW

File: fialyzer.opam [hunk 1]
```diff
@@ -14,6 +14,7 @@ depends: [
   "base"
   "obeam"
   "bitstring"
+  "zarith"
```
Comment (amutake) [inline fialyzer.opam]: zarith: https://github.com/ocaml/Zarith

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,13 +5,14 @@ jobs:
     docker:
       - image: ocaml/opam2:alpine-3.8-ocaml-4.07
     steps:
+      - run:
+          name: Install System Deps
+          # NOTE: You must install git and openssh before checkout, otherwise submodule will not be updated.
+          command: sudo apk add --update m4 make zlib-dev gmp-dev perl git openssh
```
Comment (amutake) [inline .circleci/config.yml]: zarith requires `gmp-dev` and `perl`.

**Comment by amutake** [comment]
Thank you!

**Comment by youxkei** [comment]
I think `gmp` and `perl` dependencies should be documented in `README.md`

**Pull request merged**